joplin/base/signals/janis_build_triggers.py

```diff
@@ -32,6 +31,65 @@ def trigger_build(sender, action='saved', instance=None):
         netlify_publish()
 
 
+def collect_pages(instance):
+    # does this work on page deletion? pages arent deleted right, just unpublished?
+    """
+    :param instance: the page that has been altered
+    :return: an array of global page ids -- note this may need to be a dictionary of page ids and type?
+    """
+    global_ids = []
+    wagtail_page = Page.objects.get(id=instance.id)
+    page_set = get_object_usage(wagtail_page)
+    # https://github.com/wagtail/wagtail/blob/master/wagtail/admin/models.py#L15
+    # get_object_usage will also return the wagtail_page itself
+    for page in page_set:
+        global_ids.append(Node.to_global_id(page.content_type.name, page.id))
+    global_page_id = Node.to_global_id(instance.get_verbose_name(), instance.id)
+    global_ids.append(global_page_id)
+
+    if instance.get_verbose_name() is 'Service Page' or 'Information Page':
+        guide_ids = find_pages_in_guides(instance.id)
+        global_ids.extend(guide_ids)
+
+    return global_ids
+
+
+def collect_pages_snippet(instance):
+    """
+    :param instance: the snippet that has been altered
+    :return: an array of global page ids
+    """
+    usage = instance.get_usage()
+    pages_ids = []
+    for p in usage:
+        page_id = Node.to_global_id(p.content_type.name, p.id)
+        pages_ids.append(page_id)
+    return pages_ids
+
+
+def find_pages_in_guides(changed_id):
+    """
+    Service Pages and information Pages don't know they are on Guides. So what happens if one is updated?
+    Until we know better, this will go through all our guide pages and check if the page that is changed is in
+    one of the guide's sections
+    :param changed_id: id of the page that was published / unpublished
+    :return: id of the Guide Pages that include that page
+    """
+    pages_id = []
+    all_guides = GuidePage.objects.all()
+    for g in all_guides:
+        for s in g.sections:
+            # s.value.items() is an ordered dict
+            list_of_values = list(s.value.items())
+            # the pages are the 5th tuple
+            pages = list_of_values[4][1]
```

Comment:
I'm assuming you've tried this already but just confirming that list_of_values["key_name"] doesn't work